packages/aws-cdk-lib/aws-ses/lib/configuration-set.ts

```diff
@@ -158,10 +167,23 @@ export class ConfigurationSet extends Resource implements IConfigurationSet {
       physicalName: props.configurationSetName,
     });
 
+    if (props.maxDeliveryDuration && !Token.isUnresolved(props.maxDeliveryDuration)) {
+      if (props.maxDeliveryDuration.toMilliseconds() < Duration.seconds(1).toMilliseconds()) {
+        throw new Error(`The maximum delivery duration must be greater than or equal to 5 minutes (300 seconds), got: ${props.maxDeliveryDuration.toMilliseconds()} milliseconds.`);
+      }
+      if (props.maxDeliveryDuration.toSeconds() < Duration.minutes(5).toSeconds()) {
+        throw new Error(`The maximum delivery duration must be greater than or equal to 5 minutes (300 seconds), got: ${props.maxDeliveryDuration.toSeconds()} seconds.`);
+      }
+      if (props.maxDeliveryDuration.toSeconds() > Duration.hours(14).toSeconds()) {
+        throw new Error(`The maximum delivery duration must be less than or equal to 14 hours (50400 seconds), got: ${props.maxDeliveryDuration.toSeconds()} seconds.`);
+      }
+    }
+
     const configurationSet = new CfnConfigurationSet(this, 'Resource', {
       deliveryOptions: undefinedIfNoKeys({
         sendingPoolName: props.dedicatedIpPool?.dedicatedIpPoolName,
         tlsPolicy: props.tlsPolicy,
+        maxDeliverySeconds: props.maxDeliveryDuration?.toSeconds(),
```

Comment:
What happen if it's an unresolved token and calling `toSeconds()`?

Reply:
In this case, an error is thrown.

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/lib/duration.ts#L332

If we make it accept a `number` instead of a `Duration`, it could also support tokens. However, personally, I prefer using `Duration`. What do you think we should do?